### ISSUE TITLE
Tools panel options: Unclear labelling and UI

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51655]
### Description

The Tools panel options are used, for example, to toggle the visibility of additional typographic controls. For example, clicking the `Typography options` ellipsis button in the Typography panel shows a dropdown with a list of some typography settings:

<img width="327" alt="Screenshot 2023-06-14 at 11 34 00" src="https://github.com/WordPress/gutenberg/assets/1682452/51ec113d-9f85-4971-8a82-e6a1675378f8">

As a user, it's not clear to me what the difference between 'defaults' and 'custom' is. I'm not even sure what these controls do:

**Default settings:**
- At first, the 'default' items are disabled. As a user, I wonder why these controls are there. Why do I need to know there are 'default' items? The default settings are already visible in the Typography panel. Why are they listed also in this dropdown menu?
- Only after a setting has been changed, the default items can be interacted with and only now I can understand they're meant to 'reset' any custom value previously set.

<img width="258" alt="Screenshot 2023-06-19 at 14 45 52" src="https://github.com/WordPress/gutenberg/assets/1682452/31600c8b-a1d9-49b9-9152-90d6c9f03c2a">

I'm not sure this is the best user experience, as the purpose of these controls is very unclear and needs to be 'discovered'.
When these default controls are enabled, the visible text and the aria-label mismatch. For example:
- Visible text: `Font size RESET`
- aria-label: `Reset Font size`

The mismatch will likely make speech recognition software users unable to voice the correct command because they will try to use the visible text, while the actual accessible name is different. Also, the aria-label doesn't add any important information: it just uses the same words, in a different order.

**Tools:**
As a user, it's not clear to me what these options do. There's no visible text to actually tell users these controls will make more Typography settings appear in the UI. The 'title' of this section only says 'Tools'. I would expect that by clicking, for example, 'Line height' I would be able to actually set the line height. Instead, that only reveals the line height UI.
I'd think at leat the title of this section should clarify what these controls do. For example: 'Show more tools' or something along those lines.

Labeling:
For each of the Tools options, the visible text ad the aria-label mismatch. Additionally, when one of the option is checked, the visible text stays unchanged but the aria-label changes. For example:

- Visible text: Line height
- Aria label when unchecked: Show Line height
- Aria label when checked: Hide and reset Line height

Speech recognition software users will ignore the actual accessible name of the control has changes. It is very hard for them to figure out what voice command they have to issue to use these controls. 

As a sighted user, I'm not sure what these controls do. The visible text doesn't tell anything about the actual control action. These controls either _show_ the setting UI or they _hide and reset_ the setting, but the visible text is always (for example) 'Line height'. I'm not sure this is the best user experience, as users are forced to 'discover' what these controls do by a trial-and-error process. This seems less than ideal to me. Ideally, any interface control should use a visible label that clearly informs users on what the control does.

Overall, I'd suggest to remove any aria-label and change the visible text to clearly reflect what these controls do.

### Step-by-step reproduction instructions

- Go to the post editor.
- Select a paragraph block.
- In the block settings sidebar, click the `Typography options` ellipsis button in the Typography panel.
- A dropdown menu opens.
- Observe the 'Default' and 'Tools' options.
- Observe the default ones are disabled and their purpose is unclear.
- Observe the tools ones don't tell anything about what they actually do.
- Check and uncheck a few of the tools options.
- Observe the options visible text doesn't change, while the available action did change.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
